include/gmsh.tcc

```diff
@@ -270,10 +272,14 @@ void GMSH<Tdim, Tvertices>::compute_material_points(unsigned ngauss_points) {
 
       //! Make class point and store to material points
       materialpoints_.at(material_id)
-          ->add_points(std::unique_ptr<Point<Tdim>>(
-              new Point<Tdim>(element->id(), element->id() + last_global_id,
-                              pointsarray, point_volume)));
+          ->add_points(std::shared_ptr<Point<Tdim>>(new Point<Tdim>(
+              point_id, point_id + global_id, pointsarray, point_volume)));
+
+      ++point_id;
     }
+
+    //! Update global_id for next material properties
+    global_id = point_id + 1;
```

Comment:
Since `point_id` has been pre-incremented, we don't need to update the `global_id` by +1?

Reply:
I change the name to `global_id_prev` and this talking when we have more than one material, we have the next `global_id` by continuing the number stored in `global_id_prev`